shared/src/metabase/mbql/util.cljc

```diff
@@ -624,23 +624,6 @@
                            max-results)]
     (safe-min mbql-limit constraints-limit)))
 
-;; TODO -- This seems like it would be easily confused with
-;; [[metabase.driver.sql.query-processor/source-query-alias]]. Joins are REQUIRED to have aliases anyway
-(def ^:private default-join-alias "source")
-
-(s/defn deduplicate-join-aliases :- mbql.s/Joins
-  "Make sure every join in `:joins` has a unique alias. If a `:join` does not already have an alias, this will give it
-  one."
-  [joins :- [mbql.s/Join]]
-  (let [joins          (for [join joins]
-                         (update join :alias #(or % default-join-alias)))
-        unique-aliases (uniquify-names (map :alias joins))]
-    (mapv
-     (fn [join alias]
-       (assoc join :alias alias))
-     joins
-     unique-aliases)))
```

Comment:
Previously the `resolve-joins` middleware did a half-assed attempt at deduplicating join aliases (but didn't actually rewrite `:field` clauses referencing those joins). I removed that code entirely since the new middleware handles that responsibility now.

These util functions were only used in that one spot and didn't have great test coverage so I ended up removing them since they're no longer used